src/arena.rs

```diff
@@ -166,6 +166,12 @@ impl<T> Arena<T> {
         self.storage.capacity()
     }
 
+    /// Reserve capacity for at least `additional` more elements to be inserted
+    #[inline(always)]
+    pub fn reserve(&mut self, additional: usize) {
+        self.storage.reserve(additional)
```

Comment:
Do you think that we should do some math on `additional` to take into account the existing free slots in the arena?

It'd be something along the lines of
```rust
let currently_free = self.len.saturating_sub(self.storage.len());
let to_reserve = additional.saturating_sub(currently_free);
self.storage.reserve(to_reserve);
```

Reply:
Yeah, make sense.